runtime/common/src/precompile/mod.rs

```diff
@@ -33,12 +33,13 @@ where
 		address: H160,
 		input: &[u8],
 		target_gas: Option<usize>,
+		context: &Context,
 	) -> Option<core::result::Result<(ExitSucceed, Vec<u8>, usize), ExitError>> {
-		EthereumPrecompiles::execute(address, input, target_gas).or_else(|| {
+		EthereumPrecompiles::execute(address, input, target_gas, context).or_else(|| {
 			if address == H160::from_low_u64_be(PRECOMPILE_ADDRESS_START) {
```

Comment:
maybe move the filter check to here?

Reply:
We might need custom configurable allowed list for other precompiles, to support non-predeployed contracts. Like the NFT contract for linkdrop.